.github/workflows/continuous-integration.yml

```diff
@@ -7,28 +7,72 @@ on:
   pull_request:
 
 jobs:
-  ci:
-    name: Continuous Integration
+  linting:
+    name: Linting
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Project
         uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4
-      - name: Setup Node
+
+      - name: Add ESLint Problem Matcher
+        run: echo "::add-matcher::.github/problemMatchers/eslint.json"
+
+      - name: Setup Node & Install Dependencies
         uses: josh-development/.github/setup-node@main
         with:
           turbo: true
-      - name: Add ESLint Problem Matcher
-        run: echo "::add-matcher::.github/problemMatchers/eslint.json"
+
       - name: Run ESLint
         run: yarn lint --fix=false
+
+  building:
+    name: Compile Source Code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Project
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4
+
       - name: Add TypeScript Problem Matcher
         run: echo "::add-matcher::.github/problemMatchers/tsc.json"
-      - name: Compile Source Code
-        run: yarn build --cache-dir=./.turbo
+
+      - name: Setup Node & Install Dependencies
+        uses: josh-development/.github/setup-node@main
+        with:
+          turbo: true
+
+      - name: Build Code
+        run: yarn build
+
+  docs:
+    name: Generate Documentation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Project
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4
+
+      - name: Setup Node & Install Dependencies
+        uses: josh-development/.github/setup-node@main
+        with:
+          turbo: true
+
       - name: Generate Documentation
-        run: yarn docs --cache-dir=./.turbo
+        run: yarn docs
```

Comment:
Afaik the josh-development/setup-node depends on a .turbo directory for the cache

Reply:
Possibly resolved by setting the [`TURBO_CACHE_DIR`](https://turbo.build/repo/docs/reference/command-line-reference/run#--cache-dir) env in the https://josh-development/.github/blob/main/setup-node/action.yml file.